hs_core/views/__init__.py

```diff
@@ -677,5 +677,8 @@ def _set_resource_sharing_status(request, user, resource, is_public):
         if user.uaccess.can_change_resource_flags(resource):
             resource.raccess.public = is_public
             resource.raccess.save()
+            # set isPublic metadata AVU isPublic accordingly
+            istorage = IrodsStorage()
+            istorage.setAVU(resource.short_id, "isPublic", str(is_public))
```

Comment:
The error handling is already done in setAVU() with the iRODS session run, A SessionException will be raised if there is any issue with setting iRODS metadata, so there is no need to do error handling here. In terms of creating a helper method on raccess just to combining updating public flag, saving the flag, and setting isPublic iRODS metadata, I see your point, but I am not sure whether we should mix in iRODS metadata update piece into raccess. I feel it is better to do it outside raccess. @pkdash @alvacouch what do you think?